assets/src/scripts/post-list-filters.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * Toggles the post list filter container.
+ *
+ * @param {HTMLElement} filterButton - The button used to toggle the filter container.
+ * @param {HTMLElement} filterContainer - The filter container.
+ */
+const toggleFilterContainer = ( filterButton, filterContainer ) => {
+	/**
+	 * Click event handler for the filter button.
+	 */
+	const clickHandler = () => {
+		const isHidden = ! filterContainer.classList.contains( 'post-list-filter__container--open' );
+		filterContainer.classList.toggle( 'post-list-filter__container--open' );
+
+		filterButton.innerHTML = isHidden ? 'Hide filters' : filterButton.dataset.showFiltersAppliedButtonLabel;
+	};
+
+	filterButton.addEventListener( 'click', clickHandler );
+};
+
+/**
+ * Resets the date filters.
+ *
+ * @param {HTMLElement} fromDateInput - The input element for the "from" date.
+ * @param {HTMLElement} toDateInput - The input element for the "to" date.
+ */
+const resetDateFilters = ( fromDateInput, toDateInput ) => {
+	fromDateInput.value = '';
+	toDateInput.value = '';
+};
+
+/**
+ * Walk through the form elements and reset them.
+ *
+ * @param {HTMLElement} form - The form element.
+ * @returns {void}
+ */
+const resetFormFields = form => {
+
+	// Reset all inputs on form.
+	const inputs = form.querySelectorAll( 'input[type="text"], input[type="date"]' );
+	inputs.forEach( input => {
+		input.value = '';
+	} );
+
+	// Reset all checkboxes on form.
+	const checkboxes = form.querySelectorAll( 'input[type="checkbox"]' );
+	checkboxes.forEach( checkbox => {
+		checkbox.checked = false;
+	} );
+};
+
+/**
+ * Initializes the post list filters functionality.
+ */
+const initializePostListFilters = () => {
+
+	// Controls filters container visibility toggle.
+	const filterButton = document.querySelector( '.post-list-filter__toggle' );
+	filterButton.dataset.showFiltersAppliedButtonLabel = filterButton.innerHTML;
+	const filterContainer = document.querySelector( '.post-list-filter__container' );
+	toggleFilterContainer( filterButton, filterContainer );
```

Comment:
Just as a naming nitpick - this function name sounds like it actually toggles the container. I would have named it something like 'addFilterToggleHandlers'...

Reply:
It's actually toggling the filter container, this element marked as red:

![image](https://github.com/wikimedia/shiro-wordpress-theme/assets/90911997/8b4a0cbc-9829-4816-bd25-36c35faac11d)